src/main/java/io/moderne/organizations/OrganizationStructureService.java

```diff
@@ -18,89 +21,70 @@
+    private static final String DEFAULT_REPOS_CSV = "src/main/resources/repos.csv";
     private static final String NAME_MAPPING = "id-mapping.txt";
     private static final Logger log = LoggerFactory.getLogger(OrganizationStructureService.class.getName());
 
-    private final ScmConfiguration scmConfiguration;
     private final GitRemote.Parser gitRemoteParser;
 
     @Nullable
     private final Path reposCsvPath;
 
     public OrganizationStructureService(ModerneConfiguration moderneConfiguration, GitRemote.Parser gitRemoteParser) {
         this.gitRemoteParser = gitRemoteParser;
-        this.scmConfiguration = moderneConfiguration.getScm();
         this.reposCsvPath = moderneConfiguration.getReposCsvPath();
     }
 
     public Map<String, OrganizationRepositories> readOrganizationStructure() {
         LinkedHashMap<String, OrganizationRepositories> organizations = new LinkedHashMap<>();
         Set<RepositoryInput> allRepositories = new LinkedHashSet<>();
 
-        final Map<String, String> idToNameMapping = readIdToNameMapping();
+        CsvMapper csvMapper = new CsvMapper();
+        CsvSchema bootstrapSchema = CsvSchema.emptySchema()
+                .withComments()
+                .withHeader();
 
-        InputStream inputStream;
-        try {
-            if (reposCsvPath == null) {
-                inputStream = new ClassPathResource(DEFAULT_REPOS_CSV).getInputStream();
-            } else {
-                inputStream = new FileInputStream(reposCsvPath.toFile());
-            }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        File inputFile;
+        if (reposCsvPath == null) {
+            inputFile = new File(DEFAULT_REPOS_CSV);
+        } else {
+            inputFile = reposCsvPath.toFile();
         }
 
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
-            reader.readLine(); // skip header
-            reader.lines()
-                    .forEach(line -> {
-                        String[] fields = line.split(",", Integer.MAX_VALUE);
-                        if (fields.length < 3) {
-                            throw new IllegalStateException("repos.csv lines should have at least 1 organization");
+        try (MappingIterator<ReposCsvRow> iterator = csvMapper.readerFor(ReposCsvRow.class)
+                .with(bootstrapSchema)
+                .readValues(inputFile)) {
+            while (iterator.hasNextValue()) {
+                ReposCsvRow row = iterator.nextValue();
+                RepositoryInput repository = determineRepository(row.getCloneUrl(), row.getBranch());
+
+                Queue<String> orgNames = new LinkedList<>(row.getOrganizations());
+                if (orgNames.isEmpty()) {
+                    throw new IllegalStateException("repos.csv lines should have at least 1 organization");
+                } else if (orgNames.contains("")) {
```

Comment:
I know I said I would do this as a second PR, but the previous consuming logic handles it differently, and it felt odd to write a slightly different bug, to remove later.